tools/engine_tool/lib/src/gn_utils.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:process_runner/process_runner.dart';
+
+import 'environment.dart';
+import 'proc_utils.dart';
+
+/// Information about a test build target.
+final class TestTarget {
+  /// Construct a test target.
+  TestTarget(this.label, this.executable);
+
+  /// The build target label.
+  final String label;
+
+  /// The executable file produced after the build target is built.
+  final File executable;
+
+  @override
+  String toString() {
+    return 'target=$label executable=${executable.path}';
+  }
+}
+
+/// Returns test targets for a given build directory.
+Future<Map<String, TestTarget>> findTestTargets(
+    Environment environment, Directory buildDir) async {
+  final Map<String, TestTarget> r = <String, TestTarget>{};
+  final List<String> getLabelsCommandLine = <String>[
+    'gn',
```

Comment:
The `gn` binary is in the engine tree, and we can just give the absolute path here rather than relying on it being on the `PATH` in the environment. It also needs the `.exe` when on Windows. Something like:
```dart
p.join(environment.engine.srcDir.path, 'flutter', 'third_party', 'gn', 'gn$exe')
```

Reply:
Done.